src/clj/fluree/db/reasoner/resolve.cljc

```diff
@@ -0,0 +1,93 @@
+(ns fluree.db.reasoner.resolve
+  (:require [fluree.db.constants :as const]
+            [fluree.db.flake :as flake]
+            [fluree.db.query.fql.parse :as q-parse]
+            [fluree.db.query.fql.parse :as parse]
+            [fluree.db.query.exec.where :as exec-where]
+            [fluree.json-ld :as json-ld]
+            [fluree.db.query.fql :as fql]
+            [fluree.db.util.log :as log]))
+
+#?(:clj (set! *warn-on-reflection* true))
+
+;; namespace to find all rules that should be enforced
+
+
+(defn result-summary
+  "Generates a result summary appropriate for 'schedule' function.
+
+  Input is the original rule set, and a list/set of newly generated
+  flakes from the last rules run."
+  [rules post-flakes]
+  :TODO)
+
+(defn extract-pattern*
+  [patterns]
+  (reduce
+    (fn [acc triple-pattern]
+      (conj acc
+            (mapv
+              (fn [individual]
+                (if (contains? individual ::exec-where/var)
+                  nil
+                  (::exec-where/iri individual)))
+              triple-pattern)))
+    #{}
+    patterns))
+
+(defn extract-pattern
+  [rule-statement context]
+  (let [patterns (-> (parse/parse-where-clause rule-statement nil context)
+                     ::exec-where/patterns)]
+    (extract-pattern* patterns)))
+
+(defn flake-tests
+  "Generates 'test' flakes to test if any patterns in this rule match
+  newly generate flakes (based on post index sorting)"
+  [patterns]
+  (mapv
+    (fn [[s p o]]
+      (flake/->Flake s p o nil nil nil nil))
```

Comment:
It looks like we're building flakes directly out of pattern match maps here instead of using the normal sids/primitive values. I don't see where the flake-tests are used anywhere. How will they be used to test the inferred flakes, or, do you have a pointer to where they're used?

Reply:
@zonotope These are not used yet. This match pattern can be used in a range-scan of POST containing new flakes... if it matches then there is a newly created flake that might impact the rule, and the rule must be run (or run again). If there is no match, we can safely skip this rule.

I have it there because I started work on this, but it is an optimization - so wasn't necessary to get the core functionality working. It would have been disruptive to back it all out, so I left it for now but will come back to it.